ws/config.py

```diff
@@ -1,180 +1,232 @@
-        self.subname = subname
 
-    def parse(self, stream, context=None):
-        """
-        Parses a config file and returns a dictionary of settings
-        """
-        # TODO: convert (all?) exceptions to configargparse.ConfigFileParserException
-        cf = configfile.ConfigFile(stream, inherit_options=True, safe_calls=True, interpolation=True)
-
-        try:
-            _arg = "--" + self.top_level_arg
-            _i = context.index(_arg)
-            top_level = context[_i + 1]
-        except (ValueError, IndexError):
-            top_level = None
-
-        if top_level is None:
-            top_level = cf[self.top_level_arg]
-            if not top_level:
-                raise configargparse.ConfigFileParserException("top-level parameter '{}' not found")
-        return cf(top_level, self.subname).get_options()
-
-    def serialize(self, items):
-        """
-        Does the inverse of config parsing by taking parsed values and
-        converting them back to a string representing config file contents.
+# ==============
+# Parser classes
+# ==============
 
-        :param items: an ``OrderedDict`` with items to be written to the config file
-        :returns: contents of config file as a string
-        """
-        raise NotImplementedError
-
-    def get_syntax_description(self):
-        return ""
+class ConfigParser(configparser.ConfigParser):
+    """Drop-in replacement for :py:class:`configparser.Configparser`."""
 
-class Defaults(collections.UserDict):
-    def __init__(self):
+    def __init__(self, configfile):
         super().__init__()
-        project_name = "wiki-scripts"
-
-        # global arguments
-        config_dir = os.environ.get("XDG_CONFIG_HOME", os.path.expanduser("~/.config"))
-        self.data["config"] = os.path.join(config_dir, "{0}/{0}.conf".format(project_name))
-
-        cache_dir = os.environ.get("XDG_CACHE_HOME", os.path.expanduser("~/.cache"))
-        self.data["cache_dir"] = os.path.join(cache_dir, project_name)
-
-        # ArchWiki specifics
-        self.data["site"] = "ArchWiki"
-        # NOTE: These depend on site=ArchWiki, but (config)argparse does not support conditional defaults
-        # TODO: document the behaviour properly; also emphasize that --help does not
-        #       take defaults from the config file (which is just right)
-        self.data["api_url"] = "https://wiki.archlinux.org/api.php"
-        self.data["index_url"] = "https://wiki.archlinux.org/index.php"
-
-# strict string to bool conversion
-def argtype_bool(value):
-    if isinstance(value, bool):
-        return value
-    string = str(value).lower()
-    if string in ["1", "true", "yes"]:
+        self.configfile = configfile
+
+    def fetch_section(self, section=None, to_list=True):
+        """
+        Fetches a specific section from a config file.
+
+        :param section: section name for fetching.
+        :param to_list: defines the format of the returned value (see below).
```

Comment:
Use `:param str section:` and `:param bool to_list:` since the params have an expected type.

Reply:
Done.

Reply:
I don't see the new code, you probably did not push it yet :)